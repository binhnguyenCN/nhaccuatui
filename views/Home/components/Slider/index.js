import React from 'react';

import styles from './styles.module.scss';

const Slider = () => (
  <div className={styles['slider-wrapper']}>
    <h1>SLIDER</h1>
  </div>
);
export default Slider;
