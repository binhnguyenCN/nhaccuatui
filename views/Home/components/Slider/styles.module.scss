.slider-wrapper {
  height: 15rem;
  background-color: skyblue;
  margin-bottom: 20px;
}
