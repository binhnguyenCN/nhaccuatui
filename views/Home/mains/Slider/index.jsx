// libs
import classnames from 'classnames';
import React from 'react';
import { Carousel } from 'antd';
// others
import styles from './styles.module.scss';

const Slider = ({ banners }) => (
  <div className={classnames(styles['slider-wrapper'])}>
    <div className={classnames(styles['slider-inner'])}>
      <Carousel autoplay effect="fade">
        {banners.map((banner) => (
          <div key={banner && banner.key}>
            <img
              src={banner && banner.thumbnail}
              alt="banner"
              width={1200}
              height={419}
            />
          </div>
        ))}
      </Carousel>
    </div>
  </div>
);

export default Slider;
