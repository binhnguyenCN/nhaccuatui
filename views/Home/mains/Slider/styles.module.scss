.slider-wrapper {
  width: 1200px;
  height: 419px;
  background-color: rgb(135 207 235 / 44.1%);
  margin-bottom: 50px;
  align-self: center;

  .slider-inner {
    margin: 0 auto;
    width: 1200px;
    height: 100%;

    img {
      width: 1200px;
      height: 419px;
      object-fit: cover;
    }
  }
}
