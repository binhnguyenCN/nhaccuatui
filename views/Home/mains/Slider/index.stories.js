import React from 'react';

import Slider from './index';

export default {
  title: 'Pages/Home/main/Slider',
  component: Slider,
};
export const Template = () => <Slider />;
