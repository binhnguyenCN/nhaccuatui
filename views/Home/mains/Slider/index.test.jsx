// libs
import renderer from 'react-test-renderer';
// layouts
import Slider from '.';
// mocks
import mockData from '@/mocks/mockApi';

describe('render slider', () => {
  it('should renders correctly', () => {
    // act
    const tree = renderer
      .create(<Slider banners={mockData.bannerList} />)
      .toJSON();
    // assert
    expect(tree).toMatchSnapshot();
  });
});
