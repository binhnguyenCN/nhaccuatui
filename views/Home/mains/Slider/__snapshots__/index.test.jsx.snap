// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render slider should renders correctly 1`] = `
<div
  className="slider-wrapper"
>
  <div
    className="slider-inner"
  >
    <div
      className="ant-carousel"
    >
      <div
        className="slick-slider slick-initialized"
        dir="ltr"
      >
        <div
          className="slick-list"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={null}
          onMouseMove={null}
          onMouseUp={[Function]}
          onTouchCancel={null}
          onTouchEnd={[Function]}
          onTouchMove={null}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="slick-track"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <div
              aria-hidden={false}
              className="slick-slide slick-active slick-current"
              data-index={0}
              onClick={[Function]}
              style={
                Object {
                  "left": -0,
                  "opacity": 1,
                  "outline": "none",
                  "position": "relative",
                  "transition": "opacity 500ms ease, visibility 500ms ease",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="banner"
                    height={419}
                    src="https://avatar-ex-swe.nixcdn.com/slideshow/2022/04/21/7/c/3/5/1650513030786_org.jpg"
                    width={1200}
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={1}
              onClick={[Function]}
              style={
                Object {
                  "left": -0,
                  "opacity": 0,
                  "outline": "none",
                  "position": "relative",
                  "transition": "opacity 500ms ease, visibility 500ms ease",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="banner"
                    height={419}
                    src="https://avatar-ex-swe.nixcdn.com/slideshow/2022/04/20/3/8/3/7/1650452201449_org.jpg"
                    width={1200}
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={2}
              onClick={[Function]}
              style={
                Object {
                  "left": -0,
                  "opacity": 0,
                  "outline": "none",
                  "position": "relative",
                  "transition": "opacity 500ms ease, visibility 500ms ease",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="banner"
                    height={419}
                    src="https://avatar-ex-swe.nixcdn.com/slideshow/2022/04/22/9/e/e/a/1650627480025_org.jpg"
                    width={1200}
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={3}
              onClick={[Function]}
              style={
                Object {
                  "left": -0,
                  "opacity": 0,
                  "outline": "none",
                  "position": "relative",
                  "transition": "opacity 500ms ease, visibility 500ms ease",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="banner"
                    height={419}
                    src="https://avatar-ex-swe.nixcdn.com/slideshow/2022/04/19/0/4/9/c/1650339220758_org.jpg"
                    width={1200}
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={4}
              onClick={[Function]}
              style={
                Object {
                  "left": -0,
                  "opacity": 0,
                  "outline": "none",
                  "position": "relative",
                  "transition": "opacity 500ms ease, visibility 500ms ease",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="banner"
                    height={419}
                    src="https://avatar-ex-swe.nixcdn.com/slideshow/2022/04/19/0/4/9/c/1650365706665_org.jpg"
                    width={1200}
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={5}
              onClick={[Function]}
              style={
                Object {
                  "left": -0,
                  "opacity": 0,
                  "outline": "none",
                  "position": "relative",
                  "transition": "opacity 500ms ease, visibility 500ms ease",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="banner"
                    height={419}
                    src="https://avatar-ex-swe.nixcdn.com/slideshow/2021/09/27/a/4/0/9/1632743135911_org.jpg"
                    width={1200}
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <ul
          className="slick-dots slick-dots-bottom"
          onMouseEnter={null}
          onMouseLeave={null}
          onMouseOver={null}
          style={
            Object {
              "display": "block",
            }
          }
        >
          <li
            className="slick-active"
          >
            <button
              onClick={[Function]}
            >
              1
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              2
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              3
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              4
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              5
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              6
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
